core/src/main/java/org/apache/iceberg/SnapshotScan.java

```diff
@@ -144,7 +144,9 @@ public CloseableIterable<T> planFiles() {
                   .scanMetrics(ScanMetricsResult.fromScanMetrics(scanMetrics()))
                   .metadata(metadata)
                   .build();
-          context().metricsReporter().report(scanReport);
+          context()
```

Comment:
nit: What about an explicit for each in this case? I am not sure such statements are easy to read since they are split on multiple lines.

```
for (MetricsReporter reporter : context().metricsReporters()) {
  reporter.report(scanReport);
}
```